endpoints-frameworks-v2/backend/src/main/java/com/example/echo/Echo.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not  use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.example.echo;
+
+import com.google.api.server.spi.auth.EspAuthenticator;
+import com.google.api.server.spi.auth.common.User;
+import com.google.api.server.spi.config.Api;
+import com.google.api.server.spi.config.ApiMethod;
+import com.google.api.server.spi.config.ApiNamespace;
+import com.google.api.server.spi.config.AuthLevel;
+
+/** The Echo API which Endpoints will be exposing. */
+@Api(
+    name = "echo",
+    version = "v1",
+    namespace =
+      @ApiNamespace(
+        ownerDomain = "echo.example.com",
+        ownerName = "echo.example.com",
+        packagePath = ""
+      )
+    )
+public class Echo {
+  /**
+   * Echoes the received message back.
+   *
+   * Note that name is specified and will override the default name of "{class name}.{method
+   * name}". For example, the default is "echo.echo".
+   *
+   * Note that httpMethod is not specified. This will default to a reasonable HTTP method
+   * depending on the API method name. In this case, the HTTP method will default to POST.
+   */
+  @ApiMethod(name = "echo")
+  public Message echo(Message message) {
+    return message;
+  }
+
+  /**
+   * Gets the authenticated user's email. If the user is not authenticated, this will return an HTTP
+   * 401.
+   *
+   * Note that name is not specified. This will default "{class name}.{method name}". For
+   * example, the default is "echo.getUserEmail".
+   *
+   * Note that httpMethod is not required here. Without httpMethod, this will default to GET due
+   * to the API method name. httpMethod is added here for example purposes.
+   */
+  @ApiMethod(
+      httpMethod = ApiMethod.HttpMethod.GET,
+      authenticators = {EspAuthenticator.class},
+      audiences = {"488010225785-6nbu9odsh377sl0r7gp1kvsa1l68ss6u.apps.googleusercontent.com"},
```

Comment:
We definitely don't check in things like this.  YOUR-SERVICE-ACCOUNT-HERE, or better yet, get it from a setting that maven can set such as a property that maven does substitution for.